lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -37,237 +39,1042 @@
+                                continue
+
+                        if bool(have['isAttached']) is not bool(want['isAttached']):
+                            del want['isAttached']
+                            attach_list.append(want)
+                            continue
+
+                        if bool(have['deployment']) is not bool(want['deployment']):
+                            dep_vrf = True
+
+            if not found:
+                if bool(want['deployment']):
+                    del want['isAttached']
+                    attach_list.append(want)
+
+        # logit(json.dumps(attach_list, indent=4, sort_keys=True))
+        return attach_list, dep_vrf
+
+
+    def update_attach_params(self, attach, vrf_name, deploy):
+        serial = ""
+        for ip, ser in self.ip_sn.items():
+            if ip == attach['ip_address']:
+                serial = ser
+
+        if attach:
+            attach.update({'fabric': self.fabric})
+            attach.update({'vrfName': vrf_name})
+            attach.update({'vlan': attach.get('vlan_id')})
+            attach.update({'deployment': deploy})
+            attach.update({'isAttached': deploy})
+            attach.update({'serialNumber': serial})
+            attach.update({'extensionValues': ""})
+            attach.update({'instanceValues': ""})
+            attach.update({'freeformConfig': ""})
+            if 'deploy' in attach:
+                del attach['deploy']
+            del attach['vlan_id']
+            del attach['ip_address']
+        else:
+            attach = dict()
+
+        return attach
+
+
+    def diff_for_create(self, want, have):
+        create = dict()
+        if not have:
+            create = want
+        else:
+            if have['vrfId'] != want['vrfId']:
+                logit("diff_for_create - VRF ID cant be updated to a different value")
+                pass
+                # module.exit_json("Can not update the VRF ID") # need to find a proper way to err
+            elif have['serviceVrfTemplate'] != want['serviceVrfTemplate'] or \
+                    have['source'] != want['source'] or \
+                    have['vrfTemplate'] != want['vrfTemplate'] or \
+                    have['vrfExtensionTemplate'] != want['vrfExtensionTemplate']:
+                create = want
+            else:
+                pass
+
+        return create
+
+
+    def update_create_params(self, create):
+
+        if not create:
+            return create
+
+        v_template = 'Default_VRF_Universal' if not create.get('vrf_template') else create['vrf_template']
+        ve_template = 'Default_VRF_Extension_Universal' if not create.get('vrf_extension_template') \
+            else create['vrf_extension_template']
+        src = None if not create.get('source') else create['source']
+        s_v_template = None if not create.get('service_vrf_template') else create['service_vrf_template']
+
+        create_upd = dict()
```

Comment:
I think this will perform better

```python
create_upd = {
    'fabric': self.fabric,
    'vrfName': create['vrf_name'],
    ....
}
```

Reply:
Yeah, much cleaner. I have abused dict.update, will change to above wherever possible